sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```diff
@@ -513,6 +514,8 @@ abstract class CastBase extends UnaryExpression with TimeZoneAwareExpression wit
   private[this] def castToTimestampWithoutTZ(from: DataType): Any => Any = from match {
     case DateType =>
       buildCast[Int](_, d => daysToMicros(d, ZoneOffset.UTC))
+    case TimestampType =>
+      buildCast[Long](_, ts => ts)
```

Comment:
Should we consider the input timestamp w/o tz as local offset. How about the chain:
input micros -> instant (UTC) -> local timestamp (UTC) = local timestamp (session TZ) -> instant -> result in micros

In your implementation, you say that timestamp w/o TZ is a timestamp in UTC, and it is not local timestamp.

Reply:
You are right. I will fix it soon. Thanks!